app/controllers/companies_controller.rb

```diff
@@ -13,8 +13,35 @@ def show
 
   def update_companies_list
     companies = GPW::Client.new.fetch_companies_list
+    companies.compact!
     companies.each { |company| Company.find_or_create_by!(company) }
+
     flash[:notice] = 'Companies list successfully updated'
     redirect_to companies_url
   end
+
+  def update_companies_details
+    companies = GPW::Client.new.fetch_gpw_companies_details
+    companies.compact!
+    companies.each do |company|
+      Company.find_by(isin: company[:isin])&.update(
```

Comment:
I would consider adding `!` or somehow handling errors. Otherwise you would not notice that the company was not updated due to model validation error.

```suggestion
      Company.find_by(isin: company[:isin])&.update!(
```